.github/workflows/helm_publish.yaml

```
@@ -35,16 +37,52 @@ jobs:
           destination-repo: k8gb-io/k8gb
           destination-branch: gh-pages
 
-      - name: Push to OCI Registry
+      - name: Install Cosign
+        uses: sigstore/cosign-installer@faadad0cce49287aee09b3a48701e75088a2c6ad
+        with:
+          cosign-release: 'v3.0.3' 
+
+      - name: Push and Sign Charts to OCI Registry
+        env:
+          COSIGN_EXPERIMENTAL: 1
         run: |
           echo ${{ secrets.CR_TOKEN }} | helm registry login ghcr.io -u ${{ github.actor }} --password-stdin
+          echo ${{ secrets.CR_TOKEN }} | cosign login ghcr.io -u ${{ github.actor }} --password-stdin
           git fetch origin gh-pages
           git checkout origin/gh-pages -- charts/
           ls -la charts/
           OCI_REPO="${OCI_REPO:-oci://ghcr.io/k8gb-io/charts}"
           for chart in charts/k8gb-*.tgz; do
             if [ -f "$chart" ]; then
-              helm push "$chart" "$OCI_REPO"
+              PUSH_OUTPUT=$(helm push "$chart" "$OCI_REPO" 2>&1)
+              DIGEST=$(echo "$PUSH_OUTPUT" | grep -oE 'sha256:[a-f0-9]{64}' | head -1)
+              if [ -n "$DIGEST" ]; then
+                # Sign using digest
+                CHART_REF="ghcr.io/k8gb-io/charts/k8gb@${DIGEST}"
+                echo "Signing with digest: $CHART_REF"
+                if cosign sign --yes "$CHART_REF"; then
+                  echo "Successfully signed: $CHART_REF"
+                else
+                  echo "Failed to sign: $CHART_REF"
+                  exit 1
+                fi
+              else
+                # Incase digest didnt worked then tag signing
+                CHART_NAME=$(basename "$chart" .tgz)
+                CHART_VERSION=$(echo "$CHART_NAME" | sed 's/k8gb-//')
+                CHART_REF="ghcr.io/k8gb-io/charts/k8gb:${CHART_VERSION}"
+                echo "Signing with tag: $CHART_REF"
+                if cosign sign --yes "$CHART_REF"; then
+                  echo "Successfully signed: $CHART_REF"
+                else
+                  echo "Failed to sign: $CHART_REF"
+                  exit 1
+                fi
+              fi              
+              echo "Verifying signature for: $CHART_REF"
+              cosign verify "$CHART_REF" \
+                --certificate-identity-regexp "https://github.com/${{ github.repository }}/.*" \
+                --certificate-oidc-issuer "https://token.actions.githubusercontent.com" || echo "Verification failed, but continuing..."
```

should we succeed without actual verification? It can hide the real issue and it might go unnoticed if we do not fail fast
```suggestion
                --certificate-oidc-issuer "https://token.actions.githubusercontent.com"
```